makefile

```
@@ -1,39 +1,95 @@
+# The `docker-compose logs -f` command is used to follow the logs of the `web` and `worker` services.
+# This command is useful for scaling your application horizontally by running multiple instances of a service.
 run-many :
 	$(MAKE) run-many-detached
 	$(MAKE) provision
 	docker-compose logs -f web worker
 
+# Start services in detached mode, provision them, and follow logs.
+# `run-detached` starts the `web` and `worker` services in detached mode.
+# `provision` runs a provisioning script, useful for initial setup tasks.
+# `docker-compose logs -f` tails the logs of `web` and `worker`, useful for monitoring the services.
+# This command is typically used for running your application in the background while still being able to monitor its output.
 run :
 	$(MAKE) run-detached
 	$(MAKE) provision
 	docker-compose logs -f web worker
 
+# Open a Bash shell in the Docker container.
+# `$(COMMAND_RUN)` is likely a variable defined elsewhere in your Makefile specifying how to run commands.
+# `--detach=false` ensures the command runs in the foreground.
+# `${IMG_NAME}` is a variable representing the name of the Docker image.
+# This command is useful for debugging or managing the application within the container.
 bash:
 	$(COMMAND_RUN) --detach=false ${IMG_NAME} /bin/bash -c "bash"
 
+# Start specified services in detached mode.
+# `docker-compose up -d` starts the services defined in the docker-compose.yml file in detached mode.
+# `web` and `worker` are the names of the services being started.
+# This command is used to run services in the background, freeing up the terminal.
 run-detached:
 	docker-compose up -d web worker
 
+# Start multiple instances of a service in detached mode.
+# `--scale worker=${NUM}` starts a specified number
+# (`${NUM}`) of instances of the `worker` service, along with the `web` service.
+# The `-d` flag indicates detached mode, running them in the background.
+# This command is particularly useful for testing the scalability and load balancing of your application.
 run-many-detached:
 	docker-compose up -d --scale worker=${NUM} web
 
+# Provision the application.
+# This command runs a specific Python module (`boptest_submit`) in the `provision` service.
+# `--no-deps` ensures that no linked service is started.
+# The `./testcases/${TESTCASE}` argument specifies the testcase to be used with the script.
+# This is commonly used for setting up or configuring the application, particularly in complex setups or multi-service environments.
 provision:
 	docker-compose run --no-deps provision python3 -m boptest_submit ./testcases/${TESTCASE}
 
-stop:
+# Stop and remove Docker containers, networks, images, and volumes.
+# `docker-compose down` stops the running containers and removes them along with their networks, images, and optionally, volumes.
+# This command is essential for cleanly shutting down your Dockerized environment and ensuring no residual resources are left.
+stop all MODRLC docker servers running:
```

Why'd you change the name of this directive? Can it just stay the same?
```suggestion
stop:
```